ospf6d/ospf6_area.c

```diff
@@ -343,40 +343,92 @@ void ospf6_area_disable(struct ospf6_area *oa)
 }
 
 
-void ospf6_area_show(struct vty *vty, struct ospf6_area *oa)
+void ospf6_area_show(struct vty *vty, struct ospf6_area *oa,
+		     json_object *json_areas, bool use_json)
 {
 	struct listnode *i;
 	struct ospf6_interface *oi;
 	unsigned long result;
-
-	if (!IS_AREA_STUB(oa))
-		vty_out(vty, " Area %s\n", oa->name);
-	else {
-		if (oa->no_summary) {
-			vty_out(vty, " Area %s[Stub, No Summary]\n", oa->name);
-		} else {
-			vty_out(vty, " Area %s[Stub]\n", oa->name);
+	json_object *json_area = NULL;
+	json_object *array_interfaces = NULL;
```

Comment:
It is better to set it to NULL to avoid invalid access. All other components are doing the same way.

Reply:
Please explain how dereferencing an initialized NULL pointer, which will crash the program but cannot be caught at compile time, is superior to dereferencing pointer address defined by garbage stack data, which can be caught at compile time.

>  All other components are doing the same way.

Like what?

Reply:
Yes, I understand your point. I got confused. I thought that it would give an unitialized warning. But it won't as you have already explained in the first comment (gets assaigned if used). Thanks